<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Notepad</class>
 <widget class="QMainWindow" name="Notepad">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Notepad</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="CodeEditor" name="editor"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionFindReplace"/>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionLineNumbers"/>
    <addaction name="actionFold"/>
    <addaction name="actionUnfoldAll"/>
   </widget>
   <widget class="QMenu" name="menuXML">
    <property name="title">
     <string>XML</string>
    </property>
    <addaction name="actionSyntaxHighlight"/>
    <addaction name="actionXMLSyntax"/>
    <addaction name="actionXMLSchema"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuXML"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>Clear the buffer</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open a file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save buffer to current file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save as</string>
   </property>
   <property name="toolTip">
    <string>Select a file and save the buffer there</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionFindReplace">
   <property name="text">
    <string>Find and Replace</string>
   </property>
   <property name="toolTip">
    <string>Find and replace text using regular expressions</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>Undo</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>Redo</string>
   </property>
  </action>
  <action name="actionLineNumbers">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Line numbers</string>
   </property>
   <property name="toolTip">
    <string>Show/hide line numbers</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="actionSyntaxHighlight">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Syntax highlighting</string>
   </property>
   <property name="toolTip">
    <string>Highlight XML nodes, attributes, etc.</string>
   </property>
  </action>
  <action name="actionFold">
   <property name="text">
    <string>Fold/Unfold</string>
   </property>
   <property name="toolTip">
    <string>Fold selected lines or unfold on current line</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+/</string>
   </property>
  </action>
  <action name="actionXMLSyntax">
   <property name="text">
    <string>Syntax check</string>
   </property>
   <property name="toolTip">
    <string>Check if XML is well-formed</string>
   </property>
  </action>
  <action name="actionXMLSchema">
   <property name="text">
    <string>Check with schema</string>
   </property>
   <property name="toolTip">
    <string>Select schema and validate XML</string>
   </property>
  </action>
  <action name="actionUnfoldAll">
   <property name="text">
    <string>Unfold All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+/</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>codeeditor.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
